spec/arrays.dd

```diff
@@ -490,16 +490,35 @@ a[] = b[] + 4;
 
         $(P A vector operation is indicated by the slice operator appearing
         as the left-hand side of an assignment or an op-assignment expression.
-        The right-hand side can be an expression consisting either of an array
-        slice of the same length and type as the left-hand side or a scalar expression
-        of the element type of the left-hand side, in any combination.
-        )
+        The right-hand side can be certain combinations of:)
+
+        * An array $(GLINK2 expression, SliceExpression) of the same length
+          and type as the left-hand side
+        * A scalar expression of the same element type as the left-hand side
+
+        $(P The following operations are supported:)
 
+        * Unary: `-`, `~`
+        * Add: `+`, `-`
+        * Mul: `*`, `/`, `%`,
+        * Bitwise: `^`, `&`, `|`
+        * Pow: `^^`
+
+$(SPEC_RUNNABLE_EXAMPLE_RUN
 ---
-T[] a, b, c;
-...
-a[] -= (b[] + 4) * c[];
+int[3] a = 0;
+int[] b = [1, 2, 3];
+
+a[] += 10 - (b[] ^^ 2);
+assert(a == [9, 6, 1]);
 ---
+)
+
+        $(NOTE In particular, an expression using
+        $(GLINK2 expression, ConditionalExpression),
+        $(DDSUBLINK spec/expression, logical_expressions, logical expressions),
+        $(GLINK2 expression, CmpExpression),
+        concatenation `~` or a function call is *not* a vector op.)
```

Comment:
The reason for this is because those operations are defined on slices as well as the element type, so it would be ambiguous if vector ops allowed these.

Reply:
E.g. unfortunately `slice ? 0 : 1`, `slice && 1` as well as more justified expressions like `s1 < s2` are legal.